lib/pcg/include/pcg/machine_view.h

```diff
@@ -42,4 +42,44 @@ MachineView make_1d_machine_view(device_id_t start, size_t interval_size);
 VISITABLE_STRUCT(::FlexFlow::MachineView, start, rect);
 MAKE_VISIT_HASHABLE(::FlexFlow::MachineView);
 
+MAKE_TYPEDEF_HASHABLE(::FlexFlow::DeviceID);
+MAKE_TYPEDEF_PRINTABLE(::FlexFlow::DeviceID, "DeviceID");
+
+namespace fmt {
+
+template <>
+struct formatter<::FlexFlow::DeviceType> : formatter<string_view> { 
+  template <typename FormatContext>
+  auto format(::FlexFlow::DeviceType d, FormatContext& ctx) const -> decltype(ctx.out()) {
+    using ::FlexFlow::DeviceType;
+
+    string_view name = "unknown";
+    switch (d) {
+      case DeviceType::GPU: name = "GPU"; break;
+      case DeviceType::CPU: name = "CPU"; break;
+    }
+    return formatter<string_view>::format(name, ctx);
+  } 
+};
+
+};
+
+namespace std {
+template <>
+struct hash<::FlexFlow::StridedRectangle> {
+  size_t operator()(::FlexFlow::StridedRectangle const &) const; 
+};
+
+template <>
+struct hash<::FlexFlow::MachineView> {
+  size_t operator()(::FlexFlow::MachineView const &) const;
+};
+
+template <>
+struct hash<::FlexFlow::MachineResource> {
+  size_t operator()(::FlexFlow::MachineResource const &) const;
+};
+}; // namespace std
```

Comment:
No semicolon after namespace close, and we're generally dropping the namespace comments